### PR TITLE
Beam Website Updates for 2.58.0 Release

### PR DESCRIPTION
**Pull request by jrmccluskey**
Updates the Beam website with the 2.58.0 release blog post and download links.

DO NOT MERGE until 2.58.0 is approved 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: website/www/site/content/en/blog/beam-2.58.0.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+limitations under the License.
+-->
+
+We are happy to present the new 2.58.0 release of Beam.
+This release includes both improvements and new functionality.
+See the [download page](/get-started/downloads/#2570-2024-07-22) for this release.
+
+<!--more-->
+
+For more information on changes in 2.57.0, check out the [detailed release notes](https://github.com/apache/beam/milestone/22).
```
Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:28]:
```suggestion
For more information about changes in 2.58.0, check out the [detailed release notes](https://github.com/apache/beam/milestone/22).
```

File: website/www/site/content/en/blog/beam-2.58.0.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+* Support for [Solace](https://solace.com/) source (`SolaceIO.Read`) added (Java) ([#31440](https://github.com/apache/beam/issues/31440)).
+
+## New Features / Improvements
+
+* Multiple RunInference instances can now share the same model instance by setting the model_identifier parameter (Python) ([#31665](https://github.com/apache/beam/issues/31665)).
+* Added options to control the number of Storage API multiplexing connections ([#31721](https://github.com/apache/beam/pull/31721))
+* [BigQueryIO] Better handling for batch Storage Write API when it hits AppendRows throughput quota ([#31837](https://github.com/apache/beam/pull/31837))
+* [IcebergIO] All specified catalog properties are passed through to the connector ([#31726](https://github.com/apache/beam/pull/31726))
+* Removed a 3rd party LGPL dependency from the Go SDK ([#31765](https://github.com/apache/beam/issues/31765)).
```
Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:40]: Is it worth writing out LGPL, or will users know what this is?

Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:40]:
```suggestion
* Removed a third-party LGPL dependency from the Go SDK ([#31765](https://github.com/apache/beam/issues/31765)).
```

File: website/www/site/content/en/blog/beam-2.58.0.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+* Support for MapState and SetState when using Dataflow Runner v1 with Streaming Engine (Java) ([[#18200](https://github.com/apache/beam/issues/18200)])
```
Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:41]:
```suggestion
* Support for `MapState` and `SetState` when using Dataflow Runner v1 with Streaming Engine (Java) ([[#18200](https://github.com/apache/beam/issues/18200)])
```

File: website/www/site/content/en/blog/beam-2.58.0.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+## Breaking Changes
+
+* [IcebergIO] IcebergCatalogConfig was changed to support specifying catalog properties in a key-store fashion ([#31726](https://github.com/apache/beam/pull/31726))
```
Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:45]:
```suggestion
* [IcebergIO] `IcebergCatalogConfig` was changed to support specifying catalog properties in a key-store fashion ([#31726](https://github.com/apache/beam/pull/31726))
```

File: website/www/site/content/en/blog/beam-2.58.0.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+## Bugfixes
```
Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:47]:
```suggestion
## Bug fixes
```

**Review by rszper — APPROVED** [review]

File: website/www/site/content/en/blog/beam-2.58.0.md (hunk 2 above)
Comment (jrmccluskey) [reply website/www/site/content/en/blog/beam-2.58.0.md]: LGPL should be fine here

File: CHANGES.md [hunk 6]
```diff
@@ -89,11 +89,6 @@
...
 # [2.58.0] - Unreleased
 
-## Highlights
-
-* New highly anticipated feature X added to Python SDK ([#X](https://github.com/apache/beam/issues/X)).
-* New highly anticipated feature Y added to Java SDK ([#Y](https://github.com/apache/beam/issues/Y)).
-
 ## I/Os
 
 * Support for [Solace](https://solace.com/) source (`SolaceIO.Read`) added (Java) ([#31440](https://github.com/apache/beam/issues/31440)).
```
Comment (tvalentyn) [inline CHANGES.md:94]: This sounds like a long-anticipated addition, so worth making a highlight.

File: CHANGES.md [hunk 7]
```diff
@@ -109,24 +104,11 @@
...
-* X behavior was changed ([#X](https://github.com/apache/beam/issues/X)).
 * [IcebergIO] IcebergCatalogConfig was changed to support specifying catalog properties in a key-store fashion ([#31726](https://github.com/apache/beam/pull/31726))
 
-## Deprecations
-
-* X behavior is deprecated and will be removed in X versions ([#X](https://github.com/apache/beam/issues/X)).
-
 ## Bugfixes
 
 * [BigQueryIO] Fixed a bug in batch Storage Write API that frequently exhausted concurrent connections quota ([#31710](https://github.com/apache/beam/pull/31710))
```
Comment (tvalentyn) [inline CHANGES.md:111]:
```suggestion
* [BigQueryIO] Fixed a bug in batch Storage Write API that frequently exhausted concurrent connections quota ([#31710](https://github.com/apache/beam/pull/31710))
* Fixed a logging issue where Python worker dependency installation logs sometimes were not emitted in a timely manner ([#31977](https://github.com/apache/beam/pull/31977)).
```

File: CHANGES.md (hunk 6 above)
Comment (tvalentyn) [reply CHANGES.md:94]: Thanks, let's reflect the changes here also in CHANGES.md on master after this is merged.

File: website/www/site/content/en/get-started/downloads.md [hunk 8]
```diff
@@ -96,10 +96,17 @@ versions denoted `0.x.y`.
 
 ## Releases
 
+### 2.58.0 (2024-07-22)
+Official [source code download](https://downloads.apache.org/beam/2.57.0/apache-beam-2.58.0-source-release.zip).
+[SHA-512](https://downloads.apache.org/beam/2.57.0/apache-beam-2.58.0-source-release.zip.sha512).
```
Comment (tvalentyn) [inline website/www/site/content/en/get-started/downloads.md:101]: use 2.58.0 (multiple places in URLs need a tweak)

Comment (jrmccluskey) [reply website/www/site/content/en/get-started/downloads.md]: fixed

File: website/www/site/content/en/blog/beam-2.58.0.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+We are happy to present the new 2.58.0 release of Beam.
+This release includes both improvements and new functionality.
+See the [download page](/get-started/downloads/#2570-2024-07-22) for this release.
```
Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:24]: I assume this URL will be updated before we publish.

File: website/www/site/content/en/blog/beam-2.58.0.md [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+* Added options to control the number of Storage API multiplexing connections ([#31721](https://github.com/apache/beam/pull/31721))
+* [BigQueryIO] Better handling for batch Storage Write API when it hits AppendRows throughput quota ([#31837](https://github.com/apache/beam/pull/31837))
+* [IcebergIO] All specified catalog properties are passed through to the connector ([#31726](https://github.com/apache/beam/pull/31726))
+* Removed a third-party LGPL dependency from the Go SDK ([#31765](https://github.com/apache/beam/issues/31765)).
+* Support for `MapState` and `SetState` when using Dataflow Runner v1 with Streaming Engine (Java) ([[#18200](https://github.com/apache/beam/issues/18200)])
+
+## Breaking Changes
+
+* [IcebergIO] `IcebergCatalogConfig` was changed to support specifying catalog properties in a key-store fashion ([#31726](https://github.com/apache/beam/pull/31726))
+* [SpannerIO] Added validation that query and table cannot be specified at the same time for SpannerIO.read(). Previously withQuery overrides withTable, if set ([#24956](https://github.com/apache/beam/issues/24956)).
```
Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:46]:
```suggestion
* [SpannerIO] Added validation that query and table cannot be specified at the same time for `SpannerIO.read()`. Previously `withQuery` overrides `withTable`, if set ([#24956](https://github.com/apache/beam/issues/24956)).
```

File: website/www/site/content/en/blog/beam-2.58.0.md [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+
+## Bug fixes
+
+* [BigQueryIO] Fixed a bug in batch Storage Write API that frequently exhausted concurrent connections quota ([#31710](https://github.com/apache/beam/pull/31710))
+
+## List of Contributors
+
+According to git shortlog, the following people contributed to the 2.57.0 release. Thank you to all contributors!
```
Comment (rszper) [inline website/www/site/content/en/blog/beam-2.58.0.md:54]:
```suggestion
According to git shortlog, the following people contributed to the 2.58.0 release. Thank you to all contributors!
```

**Review by rszper — APPROVED** [review]

File: website/www/site/content/en/blog/beam-2.58.0.md (hunk 9 above)
Comment (jrmccluskey) [reply website/www/site/content/en/blog/beam-2.58.0.md:24]: Yes the dates are going to be updated now that the RC has been approved

**Comment by jrmccluskey** [comment]
Going to check the links once the staging precommit is done, will merge after

**Pull request merged**